### PR TITLE
54896_show_book_detail

### PR DESCRIPTION
**Pull request by vietbq**

### CODE REVIEW

File: app/controllers/books_controller.rb [hunk 1]
```diff
@@ -1,5 +1,28 @@
...
+    else
+      get_related_books
+      get_reviews
+    end
+  end
+
+  private
+  def get_book
+    @book = Book.find_by id: params[:id]
+  end
```
Comment (NgocVu1994) [inline app/controllers/books_controller.rb]: xuống dòng

File: app/controllers/books_controller.rb [hunk 2]
```diff
@@ -1,5 +1,28 @@
 class BooksController < ApplicationController
+  before_action :get_book, only: [:show]
```
Comment (ggbond2605) [inline app/controllers/books_controller.rb]: dùng load_and_authorize_resource

File: app/controllers/books_controller.rb [hunk 3]
```diff
@@ -1,5 +1,28 @@
...
+      flash[:error] = t "flash.user.not_found_book"
+      redirect_to root_path
+    else
+      get_related_books
+      get_reviews
+    end
+  end
+
+  private
+  def get_book
```
Comment (ggbond2605) [inline app/controllers/books_controller.rb]: bỏ hàm này đi

File: app/models/book.rb [hunk 4]
```diff
@@ -9,4 +9,10 @@ class Book < ActiveRecord::Base
   validates :content, presence: true, length: {minimum: 10}
   validates :author, presence: true
   validates :number_of_pages, numericality: {only_integer: true}
+
+  enum status: [:reading, :read]
+
+  scope :get_related_books,
+    -> id, category_id, limit{where(category_id: category_id).
```
Comment (hellovietnam93) [inline app/models/book.rb]: it is necessary. Why don't use 
```@book.category.books.limit Settings.books.related_books``` 
We can get also this book.

File: app/views/books/_related.html.erb [hunk 5]
```diff
@@ -0,0 +1,24 @@
+<div class="col-xs-12 col-sm-12 col-md-12 text-center">
+  <hr>
+  <h2 class="module-title"><span><%= t "book.related" %></span></h2>
+  <% @books.each do |book| %>
+    <div class="col-md-3 col-sm-6">
+      <% if book.image.present? %>
+        <%= link_to image_tag(book.image, width: Settings.books.detail_width,
```
Comment (hellovietnam93) [inline app/views/books/_related.html.erb]: Write a helper method. It is used many times

File: app/views/books/show.html.erb [hunk 6]
```diff
@@ -0,0 +1,14 @@
+<div class="container-fluid">
+  <div class="row">
+    <hr>
+    <ul class="breadcrumb">
+      <li><a><li><%= link_to t(:home), root_path %></li></a></li>
```
Comment (hellovietnam93) [inline app/views/books/show.html.erb]: What is this?

File: app/views/books/show.html.erb [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+      <li><a><%= link_to t("book.category") %></a></li>
```
Comment (hellovietnam93) [inline app/views/books/show.html.erb]: ```link_to inside <a></a>```

File: app/views/books/_detail.html.erb [hunk 8]
```diff
@@ -0,0 +1,19 @@
+<div class="row">
+  <div class="col-xs-12 col-sm-6 col-md-5">
+    <% if @book.image.present? %>
+      <%= image_tag @book.image, width: Settings.books.detail_width,
```
Comment (hellovietnam93) [inline app/views/books/_detail.html.erb]: write helper method

File: app/helpers/books_helper.rb [hunk 9]
```diff
@@ -0,0 +1,9 @@
+module BooksHelper
+  def get_image book
+    if book.image.present? 
+      @image_url = @book.image
+    else
+      @image_url = Settings.books.url
+    end
+  end
+end
```
Comment (hellovietnam93) [inline app/helpers/books_helper.rb]: end file

File: app/helpers/books_helper.rb [hunk 10]
```diff
@@ -0,0 +1,9 @@
+module BooksHelper
+  def get_image book
+    if book.image.present?
```
Comment (hellovietnam93) [inline app/helpers/books_helper.rb]: @image_url = book.image.present? ? @book.image : Settings.books.url

File: app/views/books/_book.html.erb [hunk 11]
```diff
@@ -1,9 +1,7 @@
 <div class="col-md-3 col-sm-4 col-xs-6">
-  <% if book.image.present? %>
-    <%= link_to image_tag(book.image, size: Settings.books.size), book %>
-  <% else %>
-    <%= link_to image_tag(Settings.books.url, size: Settings.books.size), book %>
-  <% end %>
+  <% get_image book %>
+    <%= image_tag @image_url, width: Settings.books.detail_width,
```
Comment (hellovietnam93) [inline app/views/books/_book.html.erb]: cái này để làm j nữa

File: app/views/books/_detail.html.erb [hunk 12]
```diff
@@ -0,0 +1,15 @@
+<div class="row">
+  <div class="col-xs-12 col-sm-6 col-md-5">
+    <% get_image @book %>
+    <%= image_tag @image_url, width: Settings.books.detail_width,
```
Comment (hellovietnam93) [inline app/views/books/_detail.html.erb]: de lam j nua day

File: app/views/books/_review.html.erb [hunk 13]
```diff
@@ -0,0 +1,25 @@
...
+            <div class="col-md-2 col-sm-2 text-center">
+              <%= image_tag(review.user.avatar ,
+                width: Settings.books.avatar_width, class: "img-circle") %>
+              <span><%= review.user.username%></span>
+            </div>
+            <div class="col-md-10 col-sm-10">
+              <h5><%= review.user.username%></h5>
+              <p><%= time_ago_in_words review.created_at %></p>
+              <p><%= review.content %></p>
+              <%= link_to t("book.reply") %>
```
Comment (hellovietnam93) [inline app/views/books/_review.html.erb]: link_to where ?

**Comment by hellovietnam93** [comment]
OK

File: app/controllers/books_controller.rb [hunk 14]
```diff
@@ -1,5 +1,21 @@
...
   def index
     @books = Book.paginate page: params[:page], per_page: Settings.books.page
   end
+
+  def show
+    if @book.nil?
+      flash[:error] = t "flash.user.not_found_book"
+      redirect_to root_path
+    else
+      get_reviews
```
Comment (dieunb) [inline app/controllers/books_controller.rb]: load_reviews

**Pull request merged**